lib/anaconda.pm

```diff
@@ -175,8 +175,80 @@ sub custom_blivet_add_partition {
     }
 }
 
+sub custom_add_partition {
+    # Used to add partition on Rocky custom-gui partitioning screen
+    # in Anaconda. Should be called when custom-gui is displayed and free space is selected.
+    # You can pass device type for partition (needle tagged anaconda_custom_devicetype_$devicetype should exist), size of that partition in MiBs, desired filesystem of that partition
+    # (anaconda_custom_part_fs_$filesystem should exist) and mountpoint of that partition (e. g. string "/boot").
+    my %args = (
+        devicetype => "",
+        raid1 => 0,
+        size => 0,
+        filesystem => "",
+        mountpoint => "",
+        @_
+    );
+    $args{devicetype} = "raid" if $args{raid1};
+
+    # send tab until 'add' button is selected, then press
+    # the number of tabs needed seems to depend on whether a partition has already been added
+    send_key_until_needlematch("anaconda_custom_part_add", "tab");
+    send_key "spc";
+
+    # supply the mountpoint
+    assert_screen "anaconda_custom_mountpoint";
+    type_safely $args{mountpoint};
+    send_key "tab";
+    send_key "tab";
+    send_key "tab";
+    assert_screen "anaconda_custom_size";
+    # supply the desired size
+    # if size is not provided, leave empty to use the remaining disk space
+    if ($args{size}) {
+        type_safely $args{size};
+    }
+    send_key "tab";
+    send_key "tab";
+    assert_screen "anaconda_custom_btn_add_mountpoint";
+    send_key "spc";
+
+    # if no devicetype was specified or devicetype is already selected, do nothing
+    if (($args{devicetype} && !check_screen("anaconda_custom_part_fs_$args{devicetype}_selected", 5))) {
+        # send 'tab' until the Device Type dropdown is selected
+        send_key_until_needlematch("anaconda_custom_part_devicetype_selected", "tab");
+        send_key "spc";
+        # send 'down' until the correct devicetype is selected
+        send_key_until_needlematch("anaconda_custom_part_devicetype_$args{devicetype}", "down");
+        send_key "spc";
+    }
+    # if no filesystem was specified or filesystem is already selected, do nothing
+    if ($args{filesystem} && !check_screen("anaconda_custom_part_fs_$args{filesystem}_selected", 5)) {
+        # send 'tab' until the File System dropdown is selected
+        send_key_until_needlematch("anaconda_custom_part_fs_selected", "tab");
+        send_key "spc";
+        # send 'down' until the correct filesystem is selected
+        send_key_until_needlematch("anaconda_custom_part_fs_$args{filesystem}_selected", "down");
+        send_key "spc";
+    }
+    if ($args{raid1}) {
+        # send 'tab' until Raid Level dropdown is selected
+        send_key_until_needlematch("anaconda_custom_part_raidlevel_selected", "tab");
+        send_key "spc";
+        #choose RAID level from dropdown
+        # needle name specified like this is interpreted as "anaconda_custom_part_raid__selected"
```

Comment:
This is probably a Perl question, but I wanted to call out this behavior and get some team insight. Currently the `raidl` arg is specified as an int; I suspect specifying it as a string would make this problem go away, but that feels lazy. What is the correct way to handle this behavior in this case?

Reply:
My guess without digging further is that this is intended as a boolean value (Perl doesn't have a true Boolean type) where `(int)0` = `FALSE` and all other values are `TRUE` in the `if()` statement. Looking through other test files (like the postinstall check) it seems only RAID1 is currently supported although this isn't clearly documented.

Reply:
I looked a bit further through the code and I have to say... I found a mismatch 🤔
The only place such similar parameter is used is in `disk_custom_gui_software_raid.pm:23` which uses the parameter `raidl` (most likely for raidlevel) with the integer 1 for I suppose raid level 1

This means we should either rewrite the one usage in line `disk_custom_gui_software_raid.pm:23` or might be better, for future awareness, change `raid1` in the lines of anaconda.pm 185, 191 and 233 to `raidl`,
uncomment lines 238 and 239 and comment or remove 240.